### ISSUE TITLE
Brother MFC-J3530DW

### ISSUE THREAD

**Issue by wmyrda** [stefan-langenmaier/brother-overlay#60]
I became happy owner of the shiny new MFC-J3530DW. When I hooked it up I realized I need to install it in the system ;) I would kindly ask to add ebuilds for following 

http://support.brother.com/g/b/downloadlist.aspx?c=pl&lang=pl&prod=mfcj3530dw_eu_as_cn&os=127&flang=English

Hopefully it will be as simple as copying one of other ebuilds, but I know to little about them to experiment on my own so help is welcomed

**Comment by wmyrda** [comment]
I felt lucky so I created `brother-mfcj3530dw-bin-1.0.1.ebuild` which looks as follows
```
# Copyright 1999-2018 Gentoo Foundation
# Distributed under the terms of the GNU General Public License v2
# $Id$

EAPI=6

inherit eutils rpm linux-info

MODEL="mfcj3530dw"
LPR_DLF=103001
CW_DLF=103025
BR_PR=${PR/r/}

DESCRIPTION="Brother MFC-J3530DW lpr+cupswrapper (printer) drivers"
HOMEPAGE="http://support.brother.com/g/b/downloadtop.aspx?c=us&lang=en&prod=${MODEL}_us_as"
SRC_URI="http://download.brother.com/welcome/dlf${LPR_DLF}/${MODEL}lpr-${PV}-${BR_PR}.i386.rpm
        http://download.brother.com/welcome/dlf${CW_DLF}/${MODEL}cupswrapper-${PV}-${BR_PR}.i386.rpm"
LICENSE="GPL-2+ brother-eula no-source-code"
SLOT="0"
KEYWORDS="~x86 ~amd64"
IUSE=""
RESTRICT="strip mirror"

DEPEND="net-print/cups"
RDEPEND="${DEPEND}"

DEST="/opt/brother/Printers/${MODEL}"
S="${WORKDIR}${DEST}"

pkg_setup() {
	CONFIG_CHECK=""
        if use amd64; then
                CONFIG_CHECK="${CONFIG_CHECK} ~IA32_EMULATION"
        fi

	linux-info_pkg_setup
}

src_install() {
        cd cupswrapper || die
        exeinto /usr/libexec/cups/filter
        doexe brother_lpdwrapper_${MODEL}

        exeinto ${DEST}/cupswrapper
        doexe cupswrapper${MODEL}
        insinto ${DEST}/cupswrapper
        doins brother_${MODEL}_printer_en.ppd
        insinto /usr/share/cups/model/Brother
        doins brother_${MODEL}_printer_en.ppd
        insinto /usr/share/ppd/Brother
        doins brother_${MODEL}_printer_en.ppd

        cd ../lpd || die
        exeinto ${DEST}/lpd
        doexe br${MODEL}filter filter_${MODEL}

        cd ../inf || die
        insinto ${DEST}/inf
        doins br${MODEL}rc br${MODEL}func ImagingArea PaperDimension paperinfij2
        doins -r lut

        cd "${WORKDIR}/usr/bin" || die
        into /usr
        dobin brprintconf_${MODEL}
}
```
All looked ok, but I am unable to connect to the network printer. Cups finds it but says
```
lpstat -d -l -t
system default destination: Brother_MFC-J3530DW
scheduler is running
system default destination: Brother_MFC-J3530DW
device for Brother_MFC-J3530DW: lpd://BRN3C2AF4266EE4/BINARY_P1
Brother_MFC-J3530DW accepting requests since wto, 26 cze 2018, 23:02:28
printer Brother_MFC-J3530DW now printing Brother_MFC-J3530DW-1.  enabled since wto, 26 cze 2018, 23:02:28
	Unable to locate printer "BRN3C2AF4266EE4".
```

Any idea if that could be ebuild related or any other issue may be responsible here?

**Comment by stefan-langenmaier** [comment]
Hi wmyrda,

first of all thanks for writing this ebuild. If you can create a pull request I think we can work on the file so we can add it to the repository.

Concerning the error, I think it's related with the address resolution of the printer name. Maybe you can have look into an ebuild that uses avahi (https://wiki.gentoo.org/wiki/Avahi).

I recommend to try it first with the actual IP address of the printer if this works you can replace it with <printername>.local, so in your case that will be lpd://BRN3C2AF4266EE4.local/BINARY_P1

**Comment by wmyrda** [comment]
So my scanner on that printer works fine after adding it with
`brsaneconfig4 -a name=Scanner model=Brother_MFC-J3530DW ip=192.168.101.11`
but I still may not print anything even after trying `lpd://BRN3C2AF4266EE4.local/BINARY_P1` and `lpd://192.168.101.11/BINARY_P1`. Maybe the issue here is that while managing printers and changing the driver I see two drivers with the same name? I did try both with no luck, but maybe some files are misplaced and that is why CUPS sees [two drivers](https://image.ibb.co/i1Qwc8/brother.png).

**Comment by stefan-langenmaier** [comment]
What is the error message when you try to print with the fixed IP address? Are there other error messages logged? Are there differences between the two?

**Comment by wmyrda** [comment]
Full log https://pastebin.com/QEAZGb28
What stands out is `Unable to execute ippfind utility: No such file or directory`. Maybe one of the wrappers needs have entry changed with `sed` from `ipptool` to `ippfind`? Using `qlist cups |grep ipp` I find there are `/usr/bin/ipptool` and `/usr/libexec/cups/filter/sys5ippprinter`

**Comment by stefan-langenmaier** [comment]
Hi @wmyrda 

I have found this bug report: https://bugs.gentoo.org/610086

It seems you need to install cups with the zeroconf USE-flag to have the ippfind application.

Does this resolve the problem?

**Comment by wmyrda** [comment]
@stefan-langenmaier yes. Like I said in EDIT 3 "I did manage to get printer working". Link from previous post https://imgur.com/a/ekhwj62 confirms that. I just have no idea if it is Brother driver that made it work as it says "driverless mode" ;) Other options did not really worked for me

**Comment by stefan-langenmaier** [comment]
Oops, looks like I missed your edits. :)
Well I don't think it is using the driver. Did something else fail then in the logs?
